feature-foryou/src/main/java/com/google/samples/apps/nowinandroid/feature/foryou/AuthorsCarousel.kt

```diff
@@ -57,18 +57,35 @@ import com.google.samples.apps.nowinandroid.core.designsystem.icon.NiaIcons
 import com.google.samples.apps.nowinandroid.core.designsystem.theme.NiaTheme
 import com.google.samples.apps.nowinandroid.core.model.data.Author
 import com.google.samples.apps.nowinandroid.core.model.data.FollowableAuthor
+import com.google.samples.apps.nowinandroid.core.ui.JankMetricEffect
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
 
 @Composable
 fun AuthorsCarousel(
     authors: List<FollowableAuthor>,
     onAuthorClick: (String, Boolean) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    LazyRow(
-        modifier = modifier,
-        contentPadding = PaddingValues(24.dp),
-        horizontalArrangement = Arrangement.spacedBy(24.dp)
-    ) {
+    val lazyListState = rememberLazyListState()
+
+    JankMetricEffect(lazyListState) { metricsHolder ->
+        combine(
+            snapshotFlow { lazyListState.isScrollInProgress },
+            snapshotFlow { lazyListState.firstVisibleItemIndex }
+        ) { isScrollInProgress, firstVisibleItemIndex ->
+            if (isScrollInProgress) {
+                metricsHolder.state?.addState(
+                    "ForYou:AuthorsCarousel:Scrolling",
+                    "Index=$firstVisibleItemIndex"
+                )
+            } else {
+                metricsHolder.state?.removeState("ForYou:AuthorsCarousel:Scrolling")
+            }
+        }.collect()
+    }
```

Comment:
I'm not a fan of having the tracing code so close to the UI part. It clutters the composable definitions. Can we somehow extract this so inside the composables themselves you simply call `traceLazyList(lazyListState)` and be done with it?

Reply:
Created TrackScrollJank to take care of this.